options/m_option.c

```diff
@@ -1140,6 +1140,50 @@ static int parse_double_aspect(struct mp_log *log, const m_option_t *opt,
     return parse_double(log, opt, name, param, dst);
 }
 
+static void add_double_aspect(const m_option_t *opt, void *val, double add, bool wrap)
+{
+    if (opt->defval && *(double *)opt->defval == -1) {
+        double v = VAL(val);
+        v = v + add;
+
+        if (v == -1) {
+            VAL(val) = v;
+            return;
+        }
+    }
+    return add_double(opt, val, add, wrap);
+}
+
+static void multiply_double_aspect(const m_option_t *opt, void *val, double f)
+{
+    if (opt->defval && *(double *)opt->defval == -1) {
+        double v = VAL(val);
+        v *= f;
+        if (v == -1) {
+            VAL(val) = v;
+            return;
+        }
+    }
+    multiply_double_aspect(opt, val, f);
+}
+
+static int double_set_aspect(const m_option_t *opt, void *dst, struct mpv_node *src)
+{
+    if (opt->defval && *(double *)opt->defval == -1) {
+        double val = 0.0;
+        if (src->format == MPV_FORMAT_INT64) {
+            val = src->u.int64;
+        } else if (src->format == MPV_FORMAT_DOUBLE) {
+            val = src->u.double_;
+        }
```

Comment:
I know stuff like this is duplicated. I can make it all shared if you guys prefer.